ruby/vampires.rb

```diff
@@ -0,0 +1,76 @@
+valid_age = true
+eats_garlic = true
+wants_ins = true
+result = "Results inconclusive."
+valid_name = true
+likes_sunshine = true
+
+puts "How many employees will be processed?"
+num_employees = gets.chomp 
+
+i = 1
+
+until i > num_employees.to_i
+
+	puts "What is your name?"
+	name = gets.chomp
+	name = name.downcase
+	if(name == "drake cula") || (name == "tu fang")
+		valid_name = false
+	end
+
+	puts "How old are you? What year were you born?"
+	age = gets.chomp
+	birth_year = gets.chomp
+
+	if (2016 - birth_year.to_i) != age.to_i
+		valid_age = false
+	end
+
+	puts "Our company cafeteria serves garlic bread. Should we order some for you? (y/n)"
+	garlic = gets.chomp
+
+	if(garlic == "n")
+		eats_garlic = false
+	end
+
+	puts "Would you like to enroll in the company’s health insurance? (y/n)"
+	health_ins = gets.chomp 
+
+	if(health_ins == "n")
+		wants_ins = false
+	end
+
+	allergies = ""
+
+	puts "what are you allergic to? Type 'done' when finished."
+	until allergies == "done"
+		
+		allergies = gets.chomp
+		if allergies == "sunshine"
+			likes_sunshine = false
+			break
+		end
+	end
+
+	if valid_age && (eats_garlic || wants_ins)
+		result = "Probably not a vampire."
+	end
+	if (!valid_age && (!eats_garlic || !wants_ins)) || (likes_sunshine == false)
+		result = "Probably a vampire."
+	end
+	if !valid_age && !eats_garlic && !wants_ins
+		result = "Almost certainly a vampire."
+	end
+	if !valid_name

```

Comment:
I think this should be your first check. If their name is Drac or Fang then we know we have a vampire. You can nest the rest of the checks under an else following this check. if vamp == true then no reason to check anything else.

The below is an example of handling the control flow in a more efficient computational pattern.  Also, it is important to practice nesting conditions as many control flows will require you to do so.

```
if name == vamp
      vampire = true
else
      if other conditions
      elsif other conditions
      end
end
```